drivers/pwm/pwm_stm32.c

```diff
@@ -700,6 +701,17 @@ static int pwm_stm32_init(const struct device *dev)
 	}
 #endif
 
+#ifdef IS_TIM_MASTER_INSTANCE
+	if (IS_TIM_MASTER_INSTANCE(timer)) {
+		LL_TIM_SetTriggerOutput(timer, cfg->mastermode);
+	} else {
+		if (cfg->mastermode != 0) {
+			LOG_ERR("%s: Timer does not support mastermode", dev->name);
+			return -ENOTSUP;
+		}
+	}
+#endif /* IS_TIM_MASTER_INSTANCE*/
```

Comment:
Also reject non 0 values when `IS_TIM_MASTER_INSTANCE` is not defined, as done with `IS_TIM_BREAK_INSTANCE`/`st-deadtime`?
```suggestion
#else
	if (cfg->mastermode != 0) {
		LOG_ERR("%s: Timer does not support mastermode", dev->name);
		return -ENOTSUP;
	}
#endif /* IS_TIM_MASTER_INSTANCE*/